### ISSUE TITLE
imbalance: the build system requires dependencies that are not needed for simple tasks

### ISSUE THREAD

**Issue by bansp** [freedict/tools#27]
As a test, do `make clean` in the fd-dictionaries root.
Expect either a verbose report on what gets cleaned up and where, or silence signalling error-less completion.

What a new user is going to get is a stream of reports like the following:
```
make[1]: Entering directory '/home/piotr/projects/freedict/fd-dictionaries/afr-deu'
  File "../../tools/teiaddphonetics", line 95
    attrs = "".join(f' {k}="{v}"' for k, v in attrs.items())
                                ^
SyntaxError: invalid syntax
  File "../../tools/teiaddphonetics", line 95
    attrs = "".join(f' {k}="{v}"' for k, v in attrs.items())
                                ^
SyntaxError: invalid syntax
../../tools/mk/dicts.mk:103: *** Espeak or espeak-ng not installed, please install it and proceed..  Stop.
make[1]: Leaving directory '/home/piotr/projects/freedict/fd-dictionaries/afr-deu'
```

It seems to me that the build system is too dependent on the phonetics subsystem, and that creates an imbalance.
A lot of stuff can and should work without phonetics, and the `clean` target illustrates that quite perfectly: one shouldn't need to enter the phonetics subsystem in order to clean stuff up.

I stumbled on this when I tried to add a `validate_all` target: when I put the looping functionality into `dictroot.mk`, all was fine. But when I tried to generalize that and use the already existing `validation` target defined in `dicts.mk` for single databases, it turned out that I can't if I don't first install espeak* on my laptop.

So now, the request to someone who knows how: could we please have the phonetics stuff not interfere with (= not be required for) functionality that doesn't require phonetics? I'm not talking about any kind of project-wide changes, just for some clever modification of `dicts.mk` that would allow some targets (like `clean`, but also `validate_all`, and maybe some parts of what Sebastian suggests in #11) to skip the bits that invoke TEIADDPHONETICS. I am not fluent enough in the art of Makefiling to suggest the relevant modification myself :-\

**Comment by humenda** [comment]
The error that you are seeing has nothing to do with eSpeak-NG. It is that you are getting syntax errors in the python scripts (teiaddphonetics) and due to the negative return code, the make part of the build system tries to be nice but fails here.
What version of python are you running? It looks like python 2.x.

Your error report suggests that eSpeakNG as a dependency was introduced silently and without notice, yet the readme says:

> For proper use of all our tools, Perl, eSpeakNG, Python > 3.4 are required

We can make phoneme generation optional and it is in fact fairly easy. But this has not been a priority up to now and I did not see a reason. I will have a look soon.

**Comment by bansp** [comment]
I was a bit unclear, and I guess the title was part of that (gonna modify it -- sorry and thanks).
I didn't say that the phonetics system was botched in any way. I merely took the presence of errors as a signal that the build system activates modules that it doesn't need to activate, for the purpose of cleaning up or validation.

[As a side note... I left Mannheim for a week on March 13th, carrying my close-to-pocket-size notebook with Ubuntu 16.04, with hardly any disk space left and Python 2 as the default (among other obsolete things). I would normally upgrade to 20.04 in something like May 2020, on a side, while working and living on my desktop machine at the office. But I haven't been back there since March :-\ ]
Bottom line: I have two Pythons on my system, 2.7 (default) and 3.5. I fear to reset the system default, because I need the laptop to work (major deadline upcoming at the end of the month). BUT 
--> if you could find some time to look into separating the phonetics module a bit more from the rest of the build system, I think it would be a great move; Python aside, there shouldn't be a need to activate the phonetics subsystem in any way for the purpose of e.g. validation; as it is now, even after your fix and commit for #28 , I can't validate all because the phonetics/Python errors show up;

At the risk of being boringly redundant, I will return to the fragment where you quote the Readme:

> For proper use of all our tools, Perl, eSpeakNG, Python > 3.4 are required

I hope we are not cutting off people who are only interested in, say, dictionary making, or fiddling with the XSLT. None of that should require eSpeakNG or Python > 3.4. It feels like the bar is set too high for an average Joe who would maybe like to help out -- think for example of my students from way back whom I tried to put to work on starting dictionaries, or actually any newcomer to the project: let's not overwhelm them with dependencies that they don't have any use for (yet). Let's give them the project environment (fd-dictionaries, tools), and let's have them tinker in their dictionary subdirectory, while also being able to look up some of the solutions from other dictionaries. But let's not force them to meet criteria that they don't see any use for (and whose installation itself can be a challenge).

In my mind, this ticket has actually forked into two issues now: one is imbalance, the other is setting the dependency bar too high. But solving the former helps with the latter (= modularize more strongly and demand Python > 3.4 and eSpeak only from those that want the full power).
